src/framing/plaintext.rs

```diff
@@ -342,9 +347,9 @@ impl MLSPlaintextContentType {
 /// struct {
 ///     opaque mac_value<0..255>;
 /// } MAC;
-#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
 pub(crate) struct Mac {
-    pub(crate) mac_value: Vec<u8>,
+    pub(crate) mac_value: Secret,
```

Comment:
There are two separat reasons. This can be solved differently as well, but this appeared to be the easiest right now.

The `Mac` should have the same ciphersuite/version information attached as a secret. Before the value was always converted to a `Secret` before handling it, that's difficult when this information is missing. The alternative would be to store the necessary information in the `Mac`.

Checking `Mac` validity is a timing sensitive issue in general (it was the source of some TLS attacks for example). So we want to have "constant time" `Mac` comparison. And that's something that's better implemented on the generic secret. The alternative would be to have a generic "constant time" comparison function and use it in both places.